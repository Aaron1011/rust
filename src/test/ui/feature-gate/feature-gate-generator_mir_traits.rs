// compile-fail
// edition:2018

fn is_send<T: Send>(val: T) {}

async fn dummy() {}

async fn not_send() {
    let val: *const ();
    dummy().await;
}

fn main() {
    is_send(not_send());
    //~^ ERROR  `*const ()` cannot be sent between threads safely
}
