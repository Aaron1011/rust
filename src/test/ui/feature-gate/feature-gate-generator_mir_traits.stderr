error[E0277]: `*const ()` cannot be sent between threads safely
  --> $DIR/feature-gate-generator_mir_traits.rs:14:5
   |
LL | fn is_send<T: Send>(val: T) {}
   |    -------    ---- required by this bound in `is_send`
...
LL |     is_send(not_send());
   |     ^^^^^^^ `*const ()` cannot be sent between threads safely
   |
   = help: within `impl std::future::Future`, the trait `std::marker::Send` is not implemented for `*const ()`
note: future does not implement `std::marker::Send` as this value is used across an await
  --> $DIR/feature-gate-generator_mir_traits.rs:10:5
   |
LL |     let val: *const ();
   |         --- has type `*const ()`
LL |     dummy().await;
   |     ^^^^^^^^^^^^^ await occurs here, with `val` maybe used later
LL | }
   | - `val` is later dropped here

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
